Replay module: Cannot find module './node_modules/rrweb/es/rrweb/packages/rrweb/src/entries/all.js'

### Is there an existing issue for this?

- [X] I have checked for existing issues https://github.com/getsentry/sentry-javascript/issues
- [X] I have reviewed the documentation https://docs.sentry.io/
- [X] I am using the latest SDK release https://github.com/getsentry/sentry-javascript/releases

### How do you use Sentry?

Sentry Saas (sentry.io)

### Which package are you using?

@sentry/vue

### SDK Version

7.29.0

### Framework Version

Nuxt 3.0.0

### Link to Sentry event

_No response_

### Steps to Reproduce

1. Install @sentry/tracing & @sentry/vue in Nuxt 3.0.0 project

### Expected Result

Sentry is working and Nuxt renders webpage

### Actual Result

Nuxt responds with 500 status code.

Logs:
```
Listening http://[::]:3000
[nuxt] [request error] [unhandled] [500] Cannot find module './node_modules/rrweb/es/rrweb/packages/rrweb/src/entries/all.js'
Require stack:
- /home/max/Projects/<PROJECT_NAME>/.output/server/node_modules/@sentry/replay/cjs/replay.js
- /home/max/Projects/<PROJECT_NAME>/.output/server/node_modules/@sentry/replay/cjs/integration.js
- /home/max/Projects/<PROJECT_NAME>/.output/server/node_modules/@sentry/replay/cjs/index.js
- /home/max/Projects/<PROJECT_NAME>/.output/server/node_modules/@sentry/browser/cjs/index.js
- /home/max/Projects/<PROJECT_NAME>/.output/server/node_modules/@sentry/vue/cjs/index.js
  at Module._resolveFilename (node:internal/modules/cjs/loader:946:15)  
  at Module._load (node:internal/modules/cjs/loader:787:27)  
  at Module.require (node:internal/modules/cjs/loader:1012:19)  
  at require (node:internal/modules/cjs/helpers:102:18)  
  at Object.<anonymous> (./.output/server/node_modules/@sentry/replay/cjs/replay.js:9:1)  
  at Module._compile (node:internal/modules/cjs/loader:1112:14)  
  at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)  
  at Module.load (node:internal/modules/cjs/loader:988:32)  
  at Module._load (node:internal/modules/cjs/loader:834:12)  
  at Module.require (node:internal/modules/cjs/loader:1012:19)
[nuxt] [request error] [unhandled] [500] Cannot find module './node_modules/rrweb/es/rrweb/packages/rrweb/src/entries/all.js'
Require stack:
- /home/max/Projects/<PROJECT_NAME>/.output/server/node_modules/@sentry/replay/cjs/replay.js
- /home/max/Projects/<PROJECT_NAME>/.output/server/node_modules/@sentry/replay/cjs/integration.js
- /home/max/Projects/<PROJECT_NAME>/.output/server/node_modules/@sentry/replay/cjs/index.js
- /home/max/Projects/<PROJECT_NAME>/.output/server/node_modules/@sentry/browser/cjs/index.js
- /home/max/Projects/<PROJECT_NAME>/.output/server/node_modules/@sentry/vue/cjs/index.js
  at Module._resolveFilename (node:internal/modules/cjs/loader:946:15)  
  at Module._load (node:internal/modules/cjs/loader:787:27)  
  at Module.require (node:internal/modules/cjs/loader:1012:19)  
  at require (node:internal/modules/cjs/helpers:102:18)  
  at Object.<anonymous> (./.output/server/node_modules/@sentry/replay/cjs/replay.js:9:1)  
  at Module._compile (node:internal/modules/cjs/loader:1112:14)  
  at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)  
  at Module.load (node:internal/modules/cjs/loader:988:32)  
  at Module._load (node:internal/modules/cjs/loader:834:12)  
  at Module.require (node:internal/modules/cjs/loader:1012:19)
```

This bug is mentioned in #6553, but I'm not sure if it's tracked as issue is closed and bug still appears in latest release.

I can provide what error message shows up, I think there is something wrong with `license-checker-webpack-plugin` during the production build, I will keep checking.
```
yarn run v1.22.19
$ craco build

Cannot find module '/home/runner/work/repo_name/node_modules/@sentry/replay/esm/node_modules/rrweb/package.json'
Require stack:
- /home/runner/work/repo_name/node_modules/license-checker-webpack-plugin/src/licenseUtils.js
- /home/runner/work/repo_name/node_modules/license-checker-webpack-plugin/src/index.js
- /home/runner/work/repo_name/craco.config.ts
- /home/runner/work/repo_name/node_modules/cosmiconfig-typescript-loader/dist/loader.js
- /home/runner/work/repo_name/node_modules/cosmiconfig-typescript-loader/dist/index.js
- /home/runner/work/repo_name/node_modules/@craco/craco/lib/config.js
- /home/runner/work/repo_name/node_modules/@craco/craco/scripts/build.js
```

--- a/packages/replay/rollup.npm.config.js
+++ b/packages/replay/rollup.npm.config.js
@@ -1,5 +1,3 @@
-import path from 'path';
-
 import replace from '@rollup/plugin-replace';
 
 import { makeBaseNPMConfig, makeNPMConfigVariants } from '../../rollup/index';
@@ -23,12 +21,7 @@ export default makeNPMConfigVariants(
         // set exports to 'named' or 'auto' so that rollup doesn't warn about
         // the default export in `worker/worker.js`
         exports: 'named',
-
-        // As we are inlining the rrweb dependency here, we need to ensure the nested folders are correct
-        // Without this config, you get:
-        // * build/npm/esm/node_modules/rrweb/...
-        // * build/npm/esm/packages/replay/...
-        preserveModulesRoot: path.join(process.cwd(), 'src'),
+        preserveModules: false,
       },
     },
   }),

--- a/rollup/npmHelpers.js
+++ b/rollup/npmHelpers.js
@@ -95,12 +95,6 @@ export function makeBaseNPMConfig(options = {}) {
       ...Object.keys(packageDotJSON.dependencies || {}),
       ...Object.keys(packageDotJSON.peerDependencies || {}),
     ],
-
-    // TODO `'smallest'` will get rid of `isDebugBuild()` by evaluating it and inlining the result and then treeshaking
-    // from there. The current setting (false) prevents this, in case we want to leave it there for users to use in
-    // their own bundling. That said, we don't yet know for sure that that works, so come back to this.
-    // treeshake: 'smallest',
-    treeshake: false,
   };
 
   return deepMerge(defaultBaseConfig, packageSpecificConfig, {

